Dockerfile-linux.template

```diff
@@ -202,10 +202,41 @@ RUN set -eux; \
 	# Enabling frame-pointers only makes sense for Python 3.12 and newer as those have perf profiler support
 	if rcVersion | IN("3.9", "3.10", "3.11") then "" else (
 -}}
+{{ if is_alpine then ( -}}
+		arch="$(apk --print-arch)"; \
+{{ ) else ( -}}
+		arch="$(dpkg --print-architecture)"; arch="${arch##*-}"; \
+{{ ) end -}}
 # https://docs.python.org/3.12/howto/perf_profiling.html
-# But don't enable frame-pointers on 32bit x86 due to performance drop.
 # https://github.com/docker-library/python/pull/1000#issuecomment-2597021615
-	test "$gnuArch" != 'i686-linux-gnu' && EXTRA_CFLAGS="${EXTRA_CFLAGS:-} -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"; \
+		case "$arch" in \
+{{ if is_alpine then ( -}}
+			x86_64|aarch64) \
+{{ ) else ( -}}
+			amd64|arm64) \
+{{ ) end -}}
```

Comment:
Im wondering if these should be in a jq function to make it more readable but other than that, all good

Reply:
Yeah, I was tempted to do something like [golang](https://github.com/docker-library/golang/blob/985b3b1a072020748a4846fcad475786cfd6e305/Dockerfile-linux.template#L18-L52), but couldn't decide on a good way to embed it.

I was thinking something like the following, but it'll either interrupt the mental parsing of the `RUN` if placed where it is used, or the flags will be far from their context if placed before the `RUN`.
```jq
{{
	def os_arches:
    		if is_alpine then
			{
				x86_64: " -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer",
...
```